rse-project.Rmd

```diff
@@ -21,7 +21,214 @@ small pieces in predictable places with readable names are easier to find and us
+
+TODO: "Here are three examples of opinionated project structure." Design exercise that encourages comparing these structures, what do they have in common? 
+
+[Packaging data analytical work reproducibly using R (and friends)](https://peerj.com/preprints/3192/)
+
+> "1. Organizes files using conventions in the field
+> 2. Keep data, method and outputs separate while maintaining the relationship between them."
+
+```
+├── DESCRIPTION
+├── README.md
+├── LICENSE
+├── data
+│   └── my_data.csv
+└── analysis
+    └── my_report.Rmd
+```
+
+[Good enough practices in scientific computing](https://doi.org/10.1371/journal.pcbi.1005510)
+
+```
+├── CITATION
+├── README
+├── LICENSE 
+├── requirements.txt
+├── data
+│   └──birds_count_table.csv 
+├── doc
+|   ├── notebook.md
+|   ├── manuscript.md
+|   └── changelog.txt 
+├── results
+|   └── summarized_results.csv 
+└── src
+    ├── sightings_analysis.py 
+    └── runall.py
+```
+
+https://drivendata.github.io/cookiecutter-data-science/
+
+> "Well organized code tends to be self-documenting in that the organization itself provides context for your code without much overhead."
+
+```
+├── LICENSE
+├── Makefile           <- Makefile with commands like `make data` or `make train`
+├── README.md          <- The top-level README for developers using this project.
+├── data
+│   ├── external       <- Data from third party sources.
+│   ├── interim        <- Intermediate data that has been transformed.
+│   ├── processed      <- The final, canonical data sets for modeling.
+│   └── raw            <- The original, immutable data dump.
+│
+├── docs               <- A default Sphinx project; see sphinx-doc.org for details
+│
+├── models             <- Trained and serialized models, model predictions, or model summaries
+│
+├── notebooks          <- Jupyter notebooks. Naming convention is a number (for ordering),
+│                         the creator's initials, and a short `-` delimited description, e.g.
+│                         `1.0-jqp-initial-data-exploration`.
+│
+├── references         <- Data dictionaries, manuals, and all other explanatory materials.
+│
+├── reports            <- Generated analysis as HTML, PDF, LaTeX, etc.
+│   └── figures        <- Generated graphics and figures to be used in reporting
+│
+├── requirements.txt   <- The requirements file for reproducing the analysis environment, e.g.
+│                         generated with `pip freeze > requirements.txt`
+│
+├── setup.py           <- Make this project pip installable with `pip install -e`
+├── src                <- Source code for use in this project.
+│   ├── __init__.py    <- Makes src a Python module
```

Comment:
Some details around this:

There is an advantage of either naming src the same name as the root (e..g `my_project`, the convention for python packages), or having a folder within src named `my_project`. Putting `__init__`.py within this folder means that when you install this as an editable package in your environment, your functions can now be imported as `my_project.module.function` vs `src.module.function`. The latter would cause name collisions if functions are shared between projects, while the former makes it clear where the functions are from. 

There is a ticket discussion aroudn this on the DS cookie cutter page, and it is also the way of [py-scaffold](https://github.com/pyscaffold/pyscaffoldext-dsproject) (which is a (self-proclaimed) enhancement of DS cookie cutter) with a slightly less complicated folder structure)

Reply:
Ah interesting.  Do you think this convention is worth discussing in the text?  If yes, then I might get you to write it.
I could write a similar thing for the reasons R users often put their code in `R/` , e.g. it follows the conventions of an R package and you get to use nice package dev tools.